CMakeLists.txt

```diff
@@ -634,15 +637,12 @@ endif()
 if(NGRAPH_CPU_ENABLE)
     include(cmake/external_onednn.cmake)
 endif()
-if (NGRAPH_MLIR_ENABLE)
-    include(cmake/external_mlir.cmake)
-endif()
 
-if(NGRAPH_CODEGEN_ENABLE)
+if(NGRAPH_CODEGEN_ENABLE OR NGRAPH_MLIR_ENABLE)
```

Comment:
If MLIR is enabled it should set NGRAPH_CODEGEN_ENABLE true. This extra OR condition is not needed.

Reply:
NGRAPH_CODEGEN_ENABLE and NGRAPH_MLIR_ENABLE are two different usage case of LLVM and cannot be merged yet. That needs to be done in a separate PR where codegen needs to be updated to work on current LLVM master branch. I cleared up the confusion by introducing NGRAPH_LLVM_ENABLE that is turned on if NGRAPH_CODEGEN_ENABLE or NGRAPH_MLIR_ENABLE is turned on. That makes the intention clear. If LLVM usage is enabled, build LLVM.